snap/README.md

```diff
@@ -114,35 +162,335 @@ Or by using the systemd unit name and `journalctl`:
+```
+
+#### API Gateway TLS certificate setup
+
+By default Kong is configured with an EdgeX signed TLS certificate. Client validation of this certificate requires the root CA certificate from the EdgeX instance. This file
+(`ca.pem`) can be copied from directory `$SNAP_DATA/secrets/ca`.
+
+It is also possible to install your own TLS certificate to be used by the gateway. The steps to do so are as follows:
+
+Start by provisioning a TLS certificate to use. One way to do so for testing purposes is to use the edgeca snap:
+
+```bash
+$ sudo snap install edgeca
+$ edgeca gencsr --cn localhost --csr csrfile --key csrkeyfile
+$ edgeca gencert -o localhost.cert -i csrfile -k localhost.key
+```
+
+Then install the certificate:
+```bash
+$ sudo snap set edgexfoundry env.security-proxy.tls-certificate="$(cat localhost.cert)"
+$ sudo snap set edgexfoundry env.security-proxy.tls-private-key="$(cat localhost.key)"
+```
+
+This sample certificate is signed by the EdgeCA root CA, so by specifying the Root CA certificate for validation then a connection can now be made using your new certificate:
+
+```bash
+$ curl -v --cacert /var/snap/edgeca/current/CA.pem -X GET https://localhost:8443/coredata/api/v1/ping? -H "Authorization: Bearer $TOKEN"
+```
+
+To set the certificate again, you first need to clear the current setting by setting the values to an empty string:
+
+```bash
+$ sudo snap set edgexfoundry env.security-proxy.tls-certificate=""
+$ sudo snap set edgexfoundry env.security-proxy.tls-private-key=""
+```
+
+If you are using a different server name than localhost, then it needs to be specified first using the optional tls-sni configuration setting. Example:
+
+```bash
+$ edgeca gencsr --cn server01 --csr csrfile --key csrkeyfile
+$ edgeca gencert -o server.cert -i csrfile -k server.key
+$ sudo snap set edgexfoundry env.security-proxy.tls-certificate=""
+$ sudo snap set edgexfoundry env.security-proxy.tls-private-key=""
+$ sudo snap set edgexfoundry env.security-proxy.tls-sni="server01"
+$ sudo snap set edgexfoundry env.security-proxy.tls-certificate="$(cat server.cert)"
+$ sudo snap set edgexfoundry env.security-proxy.tls-private-key="$(cat server.key)"
+
+$ curl -v --cacert /var/snap/edgeca/current/CA.pem -X GET https://server01:8443/coredata/api/v1/ping? -H "Authorization: Bearer $TOKEN"
 ```
-**Note** - Kong is currently not supported in the snap when installed on an arm64-based device, so it will be disabled on install.
 
 ## Limitations
 
 [See the GitHub issues with label snap for current issues.](https://github.com/edgexfoundry/edgex-go/issues?q=is%3Aopen+is%3Aissue+label%3Asnap)
 
+## Service environment configuration overrides
+**Note** - all of the configuration options below must be specified with the prefix: 'env.<service>.' where '<service>' is one of the following:
+
+  - core-command
+  - core-data
+  - core-metadata
+  - support-notifications
+  - support-scheduler
+  - sys-mgmt-agent
+  - security-proxy
+  - security-secret-store
```

Comment:
Didn't you change these two (proxy and secret-store) to use a `-setup` suffix?

Reply:
Added the `-setup` suffix here